src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/Parsers/ExpressionParser.cs

```diff
@@ -68,11 +68,29 @@ public Expression Parse(string[] resourceTypes, string key, string value)
         public IncludeExpression ParseInclude(string[] resourceTypes, string includeValue, bool isReversed, bool iterate, IEnumerable<string> allowedResourceTypesByScope)
         {
             var valueSpan = includeValue.AsSpan();
-            if (!TrySplit(SearchSplitChar, ref valueSpan, out ReadOnlySpan<char> originalType))
+
+            SearchParameterInfo refSearchParameter = null;
+            bool wildCard = false;
+
+            // check before split if this is a wildcard match
+            if (valueSpan.Equals("*".AsSpan(), StringComparison.InvariantCultureIgnoreCase))
+            {
+                refSearchParameter = null;
+                wildCard = true;
+            }
+
+            if (!TrySplit(SearchSplitChar, ref valueSpan, out ReadOnlySpan<char> originalType) && !wildCard)
```

Comment:
If `valueSpan == "*"` then we shouldn't need to check for the split char, so perhaps:

```suggestion
            else if (!TrySplit(SearchSplitChar, ref valueSpan, out ReadOnlySpan<char> originalType) && !wildCard)
```